### ISSUE TITLE
i64 lowering only when needed

### ISSUE THREAD

**Issue by syrusakbary** [wasmerio/wasmer-js#31]
Right now, we are doing i64 to i32 lowering always because Browsers don't currently support calling a function with a i64 signature.

However, in a few months browser WebAssembly implementations will support it (and Node as well), therefore we should check if i64 function calls are supported and in case is not, lower it.

# How we can check it?

1. Create dummy WebAssembly file with a i64 signature that returns an i64
2. Try to call it. If it succeeds, then it works otherwise we use our "polyfill lowering the i64s"

**Comment by torch2424** [comment]
This is no longer our problem, with the new functional API 😄 That being said, would be cool to make another package that does this 👍

**Comment by syrusakbary** [comment]
I just created this PR:
https://github.com/GoogleChromeLabs/wasm-feature-detect/pull/8

We can use it to detect support for BigInts and only lower the i64 when needed;

**Comment by torch2424** [comment]
Nice! @syrusakbary 😄

**Comment by syrusakbary** [comment]
The PRs are now merged, we can use `wasm-feature-detect>=1.2.0`.

```js
  import { bigInt } from "wasm-feature-detect";
 
  bigInt().then(bigIntSupported => {
    if (bigIntSupported) {
      /* bigInt support */
    } else {
      /* No bigInt support */
    }
  });
```

**Comment by torch2424** [comment]
Awesome 👍
